src/core/models/payloads/transaction.c

```diff
@@ -12,7 +12,9 @@ transaction_essence_t* tx_essence_new() {
   transaction_essence_t* es = malloc(sizeof(transaction_essence_t));
   if (es) {
     es->tx_type = TRANSACTION_PAYLOAD_ESSENCE;  // 0 to denote a transaction essence.
+    es->network_id = 0;
```

Comment:
network id is mandatory and known before creating the transaction, it should be passed by `tx_essence_new` 
`transaction_essence_t* tx_essence_new(uint64_t network_id)`

the same with the protocol version in the message
` core_message_t* core_message_new(uint8_t ver)`

Reply:
Added `network Id` and `protocol version` as functions parameters.